Apply linters to legacy protos

codingllama opened the pull request:
Applies linters to legacy protos and adds a few additional Makefile targets to make it easier to manage protos locally.

Proto linters now run in CI.

#15187

codingllama committed:
Apply linters to legacy protos

codingllama commented:
Branched from #15870.

While I was doing this I figured we might as well include Connect protos in the checks, so there is some overlap between this and #15877 - whichever gets merged last can fix the TODOs added.

codingllama committed:
Handle new folders in genproto.sh, reset gen/proto if exists

codingllama commented:
Friendly ping?

ravicious reviewed (approved):
I confirmed that it works by making some changes to the files and running `make protos/all`.

codingllama committed:
Lint and format lib/teleterm as part of protos/all

codingllama commented:
FYI, fixed TODOs related to Connect (`make protos/all` now includes Connect protos) and rebased on top of #15877.

#15877 is fully approved and on its way to be merged, so it seemed appropriate to push the fixes here.
